client/views/parameters-editor.js

```diff
@@ -30,14 +30,13 @@ module.exports = View.extend({
   template: template,
   events: {
     'click [data-hook=add-parameter]' : 'addParameter',
-    'click [data-hook=save-parameters]' : 'switchToViewMode',
     'click [data-hook=collapse]' : 'changeCollapseButtonText',
   },
   initialize: function (attrs, options) {
     var self = this;
     View.prototype.initialize.apply(this, arguments);
+    this.readOnly = attrs.readOnly ? attrs.readOnly : false;
```

Comment:
`attrs.readOnly ? attrs.readOnly : false` is always equivalent to the value of `attrs.readOnly`

Reply:
We need to check if `attrs.readOnly` is defined